From 57e70fb2939330a1dc413f018a5b5336ebe2974e Mon Sep 17 00:00:00 2001
From: Rafael Martins <rmartins@anaconda.com>
Date: Tue, 16 Jul 2024 23:10:51 +0200
Subject: [PATCH 1/6] chromium: add conda CDT include paths to include_dirs

cmake won't properly pass the include dirs to gn. we need to help it
find our headers and libraries.

---
 src/3rdparty/chromium/build/config/BUILD.gn                   | 5 +++++
 src/3rdparty/chromium/build/config/compiler/BUILD.gn          | 11 ++++++++++-
 src/3rdparty/chromium/third_party/angle/BUILD.gn              | 11 ++++++++++-
 src/3rdparty/chromium/third_party/vulkan-headers/src/BUILD.gn | 11 ++++++++++-
 4 files changed, 35 insertions(+), 3 deletions(-)

diff --git a/src/3rdparty/chromium/build/config/BUILD.gn b/src/3rdparty/chromium/build/config/BUILD.gn
index 55052f6..5e39693 100644
--- a/src/3rdparty/chromium/build/config/BUILD.gn
+++ b/src/3rdparty/chromium/build/config/BUILD.gn
@@ -190,6 +190,9 @@ config("default_libs") {
       # some extra libraries, please just add a libs = [ "foo.lib" ] to your
       # target that needs it.
     ]
+
+    lib_dirs = [ getenv("LIBRARY_LIB") ]
+
     if (current_os == "winuwp") {
       # These libraries are needed for Windows UWP (i.e. store apps).
       libs += [
@@ -213,6 +216,7 @@ config("default_libs") {
     # Targets should choose to explicitly link frameworks they require. Since
     # linking can have run-time side effects, nothing should be listed here.
     libs = []
+    lib_dirs = [ getenv("PREFIX") + "/lib" ]
   } else if (is_ios) {
     # Targets should choose to explicitly link frameworks they require. Since
     # linking can have run-time side effects, nothing should be listed here.
@@ -223,6 +227,7 @@ config("default_libs") {
       "pthread",
       "rt",
     ]
+    lib_dirs = [ getenv("PREFIX") + "/lib" ]
   }
 }

diff --git a/src/3rdparty/chromium/build/config/compiler/BUILD.gn b/src/3rdparty/chromium/build/config/compiler/BUILD.gn
index f596aa2..d6b899d 100644
--- a/src/3rdparty/chromium/build/config/compiler/BUILD.gn
+++ b/src/3rdparty/chromium/build/config/compiler/BUILD.gn
@@ -253,7 +253,16 @@ assert(!(llvm_force_head_revision && use_remoteexec && host_os != "linux"),
 # source root and might have conflicting versions of some headers) can remove
 # this and specify their own include paths.
 config("default_include_dirs") {
-  include_dirs = [
+  if (is_win) {
+    include_dirs = [ getenv("LIBRARY_INC") ]
+  } else {
+    include_dirs = [
+      getenv("PREFIX") + "/include",
+      getenv("BUILD_PREFIX") + "/" + getenv("HOST") + "/sysroot/usr/include"
+    ]
+  }
+
+  include_dirs += [
     root_gen_dir,
     "//",
   ]
diff --git a/src/3rdparty/chromium/third_party/angle/BUILD.gn b/src/3rdparty/chromium/third_party/angle/BUILD.gn
index bdd0871..1d53f63 100644
--- a/src/3rdparty/chromium/third_party/angle/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/angle/BUILD.gn
@@ -463,7 +463,16 @@ config("angle_asserts_config") {
 }

 config("angle_common_config") {
-  include_dirs = [
+  if (is_win) {
+    include_dirs = []
+  } else {
+    include_dirs = [
+      getenv("PREFIX") + "/include",
+      getenv("BUILD_PREFIX") + "/" + getenv("HOST") + "/sysroot/usr/include"
+    ]
+  }
+
+  include_dirs += [
     "src/common/base",
     "src/common/third_party/xxhash",
   ]
diff --git a/src/3rdparty/chromium/third_party/vulkan-headers/src/BUILD.gn b/src/3rdparty/chromium/third_party/vulkan-headers/src/BUILD.gn
index d5f104d..214691f 100644
--- a/src/3rdparty/chromium/third_party/vulkan-headers/src/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/vulkan-headers/src/BUILD.gn
@@ -6,7 +6,16 @@
 import("//build_overrides/vulkan_headers.gni")

 config("vulkan_headers_config") {
-  include_dirs = [ "include" ]
+  if (is_win) {
+    include_dirs = []
+  } else {
+    include_dirs = [
+      getenv("PREFIX") + "/include",
+      getenv("BUILD_PREFIX") + "/" + getenv("HOST") + "/sysroot/usr/include"
+    ]
+  }
+
+  include_dirs += [ "include" ]
   defines = []

   if (is_win) {
--
2.39.2 (Apple Git-143)
